test/gtest/solver.hpp

```diff
@@ -0,0 +1,166 @@
+                                        conv_config.stride_x,
+                                        conv_config.dilation_y,
+                                        conv_config.dialtion_x);
+
+        int n, c, h, w;
+        miopenGetConvolutionForwardOutputDim(conv_desc, &input.desc, &weights.desc, &n, &c, &h, &w);
+
+        output  = tensor<T>{static_cast<size_t>(n),
+                           static_cast<size_t>(c),
+                           static_cast<size_t>(h),
+                           static_cast<size_t>(w)};
+        ref_out = tensor<T>{static_cast<size_t>(n),
+                            static_cast<size_t>(c),
+                            static_cast<size_t>(h),
+                            static_cast<size_t>(w)};
+
+        std::fill(output.begin(), output.end(), std::numeric_limits<double>::quiet_NaN());
+        std::fill(ref_out.begin(), ref_out.end(), std::numeric_limits<double>::quiet_NaN());
+
+        auto&& handle = get_handle();
+        in_dev        = handle.Write(input.data);
+        wei_dev       = handle.Write(weights.data);
+        out_dev       = handle.Write(output.data);
+    }
+    void TearDown() override
+    {
+        if(test_skipped)
+        {
+            miopenDestroyConvolutionDescriptor(conv_desc);
+            return;
+        }
+
+        auto&& handle = get_handle();
+
+        miopen::TensorDescriptor output_desc =
+            miopen::deref(conv_desc).GetForwardOutputTensor(input.desc, weights.desc, miopenFloat);
+        ref_out = tensor<T>{output_desc.GetLengths()};
+        cpu_convolution_forward(miopen::deref(conv_desc).GetSpatialDimension(),
+                                input,
+                                weights,
+                                ref_out,
+                                miopen::deref(conv_desc).GetConvPads(),
+                                miopen::deref(conv_desc).GetConvStrides(),
+                                miopen::deref(conv_desc).GetConvDilations(),
+                                miopen::deref(conv_desc).GetGroupCount());
+
+        output.data = handle.Read<T>(out_dev, output.data.size());
+        EXPECT_FALSE(miopen::range_zero(ref_out)) << "Cpu data is all zeros";
+        EXPECT_FALSE(miopen::range_zero(output)) << "Gpu data is all zeros";
+        EXPECT_TRUE(miopen::range_distance(ref_out) == miopen::range_distance(output));
+
+        const double tolerance = 80;
+        double threshold       = std::numeric_limits<T>::epsilon() * tolerance;
+        auto error             = miopen::rms_range(ref_out, output);
+
+        EXPECT_FALSE(miopen::find_idx(ref_out, miopen::not_finite) >= 0)
+            << "Non finite number found in the CPU data";
+
+        EXPECT_TRUE(error < threshold)
+            << "Error beyond tolerance Error:" << error << ",  Threshold: " << threshold;
+
+        miopenDestroyConvolutionDescriptor(conv_desc);
+    }
+    ConvTestCase conv_config;
+    miopenConvolutionDescriptor_t conv_desc;
```

Comment:
If we use the MIOpen C++ objects instead, then you would not need to manage the lifetime of these objects.

Reply:
> If we use the MIOpen C++ objects instead, then you would not need to manage the lifetime of these objects.

@JehandadKhan , done